File: android/SwiftKotlination/features/story/src/main/kotlin/fr/jhandguy/story/model/StoryManager.kt
package fr.jhandguy.story.model

import fr.jhandguy.network.model.observer.Observer
import fr.jhandguy.network.model.observer.Result

interface StoryManagerInterface {
    suspend fun story(observer: Observer<Story>)
}

class StoryManager(private val story: Story) : StoryManagerInterface {
    override suspend fun story(observer: Observer<Story>) = observer(Result.Success(story))
}

File: ios/Features/StoryKit/StoryKit/Sources/Model/StoryManager.swift
import NetworkKit

public protocol StoryManagerProtocol {
    func story(_ observer: @escaping Observer<Story>)
}

public struct StoryManager {
    let story: Story

    public init(story: Story) {
        self.story = story
    }
}

extension StoryManager: StoryManagerProtocol {
    public func story(_ observer: @escaping Observer<Story>) {
        observer(.success(story))
    }
}
